layer/nrlf/core/validators.py

```diff
@@ -256,6 +256,32 @@ def _validate_relates_to(self, model: DocumentReference):
                 )
                 continue
 
+    def _validate_asid(self, asid_references: list):
+        """
+        Validate that the ASID provided in the document is valid
+        """
+        logger.log(LogReference.VALIDATOR001, step="ssp_asid")
+
+        if len(asid_references) > 1:
+            self.result.add_error(
+                issue_code="invalid",
+                error_code="INVALID_RESOURCE",
+                diagnostics="Multiple ASID identifiers provided. context.related must contain a single valid ASID identifier when content contains an SSP URL",
+                field="context.related",
+            )
+            return
+
+        idx, asid_reference = asid_references[0]
+        asid_value = getattr(asid_reference.identifier, "value", "")
+        if not match(r"^\d{12}$", asid_value):
+            self.result.add_error(
+                issue_code="value",
+                error_code="INVALID_IDENTIFIER_VALUE",
+                diagnostics=f"Invalid ASID value '{asid_value}'. context.related must contain a single valid ASID identifier when content contains an SSP URL",
```

Comment:
Maybe remove the SSP URL bit from this specific diagnostics message as it's longer linked directly to SSP URLs.